cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java

```diff
@@ -218,6 +258,292 @@ public static void applyTemplate(Path modulePath, String template) throws IOExce
+        String orgName = findOrg(template);
+        String version = findPkgVersion(template);
+        if (version != null) {
+            //First we will check for a bala that match any platform
+            Path balaPath = balaCache.resolve(
+                    ProjectUtils.getRelativeBalaPath(orgName, packageName, version, null));
+            String platform = findPlatform(balaPath);
+            balaPath = balaCache.resolve(
+                    ProjectUtils.getRelativeBalaPath(orgName, packageName, version, platform));
+            if (Files.exists(balaPath)) {
+                return balaPath;
+            } else {
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    public static void pullPackageFromCentral(Path balaCache, Path projectPath, String template)
+            throws CentralClientException {
+        System.setProperty(CentralClientConstants.ENABLE_OUTPUT_STREAM, "true");
+        String packageName = findPkgName(template);
+        String orgName = findOrg(template);
+        String version = findPkgVersion(template);
+        Path packagePathInBalaCache = balaCache.resolve(orgName).resolve(packageName);
+        // create directory path in bala cache
+        try {
+            createDirectories(packagePathInBalaCache);
+        } catch (IOException e) {
+            CommandUtil.exitError(exitWhenFinish);
+            throw createLauncherException(
+                    "unexpected error occurred while creating package repository in bala cache: " + e.getMessage());
+        }
+
+        for (String supportedPlatform : SUPPORTED_PLATFORMS) {
+            try {
+                Files.createDirectories(projectPath);
+                Settings settings;
+                try {
+                    settings = readSettings();
+                    // Ignore Settings.toml diagnostics in the pull command
+                } catch (SettingsTomlException e) {
+                    // Ignore 'Settings.toml' parsing errors and return empty Settings object
+                    settings = Settings.from();
+                }
+                CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
+                        initializeProxy(settings.getProxy()),
+                        getAccessTokenOfCLI(settings));
+                client.pullPackage(orgName, packageName, version, packagePathInBalaCache, supportedPlatform,
+                        RepoUtils.getBallerinaVersion(), false);
+                applyBalaTemplate(projectPath, balaCache, template);
+            } catch (PackageAlreadyExistsException e) {
+                throw new PackageAlreadyExistsException(e.getMessage());
+            } catch (CentralClientException ce) {
+                if (version == null) {
+                    outStream = System.out;
+                    if (ce.getMessage().contains("nodename nor servname provided, or not known")) {
```

Comment:
We don't need to filter the error message here. Remove this if condition and simply continue with the warning